client/src/ui/modules/world-structures/WorldStructuresMenu.tsx

```diff
@@ -297,16 +253,16 @@ const HyperStructureExtraContent = ({
   x: number;
   y: number;
 }) => {
-  const dojo = useDojo();
   const {
     account: { account },
-  } = dojo;
+  } = useDojo();
 
   const progress = useHyperstructureProgress(hyperstructureEntityId);
-
-  const latestChangeEvent = LeaderboardManager.instance(dojo).getCurrentCoOwners(hyperstructureEntityId);
-
+  const latestChangeEvent = LeaderboardManager.instance(useDojo()).getCurrentCoOwners(hyperstructureEntityId);
   const needTosetCoOwners = !latestChangeEvent && progress.percentage === 100;
+  const shares =
+    LeaderboardManager.instance(useDojo()).getAddressShares(ContractAddress(account.address), hyperstructureEntityId) ||
+    0;
```

Comment:
Could we refactor this bit to something like this
{
const dojo = useDojo(); // use dojo.account.account where necessary

const leaderboardManager = LeaderboardManager.instance(dojo);
leaderboardManager.getCurrentCoOwners
leaderboardManager.getAddressShares
}